cmd/podman/common.go

```diff
@@ -39,24 +39,45 @@ func shortID(id string) string {
 	return id
 }
 
-// checkAllAndLatest checks that --all and --latest are used correctly
-func checkAllAndLatest(c *cobra.Command, args []string, ignoreArgLen bool) error {
+// checkAllAndLatest checks that --all and --latest are used correctly. If
+// cidfile is set, also check for the --cidfile flag.
+func checkAllAndLatest(c *cobra.Command, args []string, ignoreArgLen bool, cidfile bool) error {
 	argLen := len(args)
 	if c.Flags().Lookup("all") == nil || c.Flags().Lookup("latest") == nil {
-		return errors.New("unable to lookup values for 'latest' or 'all'")
+		if !cidfile {
+			return errors.New("unable to lookup values for 'latest' or 'all'")
+		} else if c.Flags().Lookup("cidfile") == nil {
+			return errors.New("unable to lookup values for 'latest', 'all' or 'cidfile'")
+		}
+	}
+
+	specifiedAll, _ := c.Flags().GetBool("all")
+	specifiedLatest, _ := c.Flags().GetBool("latest")
+	specifiedCIDFile := false
+	if cid, _ := c.Flags().GetString("cidfile"); cid != "" {
```

Comment:
Why can't you just check for this in the same line that you lookup all and latest?

Reply:
`all` and `latest` are bools, while `cidfile` is a string. So we first need to lookup the string and then check if it's empty or not.